include/sys/range_tree.h

```diff
@@ -48,6 +48,11 @@ typedef enum zfs_range_seg_type {
 	ZFS_RANGE_SEG_NUM_TYPES,
 } zfs_range_seg_type_t;
 
+#define	ZFS_RANGE_TREE_UC_UNKNOWN		(1 << 0)
+#define	ZFS_RANGE_TREE_UC_ALLOCATED_SPACE	(1 << 1)
+#define	ZFS_RANGE_TREE_UC_FREE_SPACE		(1 << 2)
+#define	ZFS_RANGE_TREE_UC_DYN_INSTANCE		(1 << 16)
```

Comment:
Please document what these #defines mean.

Also, why is `ZFS_RANGE_TREE_UC_DYN_INSTANCE` bit 16 instead of 4?

Reply:
> Please document what these #defines mean.

Sure, I hope the added comments provide general explanation and cover the available flags.

> Also, why is `ZFS_RANGE_TREE_UC_DYN_INSTANCE` bit 16 instead of 4?

I do not think we are going to have special support for other cases, but intuitively I ended up grouping the actual use cases (the first 16 bits) and separating some technical flags. It is not a big deal to flatten it, just let me know.

Reply:
Ok, I see.  I would still flatten it as our typical convention is to make the bitfields positions sequential, even if their functions are very different.

Reply:
I see, thanks for describing the agreement.